src/main/java/edu/illinois/library/cantaloupe/cache/FilesystemCache.java

```diff
@@ -844,44 +844,34 @@ public void purgeExpired() throws CacheException {
                 }
             }
         }
-        logger.info("purgeExpired(): purging...");
 
         try {
             globalPurgeInProgress.set(true);
-            final String imagePathname = rootDerivativeImagePathname();
-            final String infoPathname = rootInfoPathname();
-
-            long imageCount = 0;
-            final File imageDir = new File(imagePathname);
-            Iterator<File> it = FileUtils.iterateFiles(imageDir, null, true);
-            while (it.hasNext()) {
-                final File file = it.next();
-                if (isExpired(file)) {
-                    try {
-                        FileUtils.forceDelete(file);
-                        imageCount++;
-                    } catch (IOException e) {
-                        logger.warn(e.getMessage());
-                    }
-                }
-            }
 
-            long infoCount = 0;
-            final File infoDir = new File(infoPathname);
-            it = FileUtils.iterateFiles(infoDir, null, true);
-            while (it.hasNext()) {
-                final File file = it.next();
-                if (isExpired(file)) {
-                    try {
-                        FileUtils.forceDelete(file);
-                        infoCount++;
-                    } catch (IOException e) {
-                        logger.warn(e.getMessage());
-                    }
-                }
-            }
-            logger.info("purgeExpired(): purged {} expired image(s) and {} " +
-                    "expired infos(s)", imageCount, infoCount);
+            final String[] pathnamesToPurge = {
+		rootSourceImagePathname(),
+		rootDerivativeImagePathname(),
+		rootInfoPathname() };
+            for (String pathname : pathnamesToPurge) {
```

Comment:
I refactored a bit to be a bit more like this: https://github.com/BossaNova/cantaloupe/blob/5fc0373a3499ddb3398ee0893f7d19e79df67d55/src/main/java/edu/illinois/library/cantaloupe/cache/FilesystemCache.java#L447